command/login_test.go

```diff
@@ -56,16 +56,6 @@ func TestLogin(t *testing.T) {
 			svcs := disco.NewWithCredentialsSource(creds)
 			svcs.SetUserAgent(httpclient.TerraformUserAgent(version.String()))
 
-			svcs.ForceHostServices(svchost.Hostname("app.terraform.io"), map[string]interface{}{
-				"login.v1": map[string]interface{}{
-					// On app.terraform.io we use password-based authorization.
-					// That's the only hostname that it's permitted for, so we can't
-					// use a fake hostname here.
-					"client":      "terraformcli",
-					"token":       s.URL + "/token",
-					"grant_types": []interface{}{"password"},
-				},
-			})
```

Comment:
Note to other reviewers: we're removing this test setup in favour of the one below, in order to test the new MOTD feature. We can't test both, and password-based authorization is no longer available for Terraform Cloud, with no current plans to reintroduce it.

Now that I'm writing this, I think we could safely remove the code which implements password-based authentication, since it's essentially inaccessible. Not critical to do in this PR in my opinion.

Reply:
Agreed; we definitely should remove a lot of cruft in this area, but this particular PR's goal is to minimize changes 👍🏻